tools/src/tester/StoryValidator.ts

```diff
@@ -0,0 +1,55 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+import { Result, StoryEvaluation, StoryFile } from "./types/eval.types";
+import * as path from "path";
+import { Ajv2019, ValidateFunction } from 'ajv/dist/2019'
+import addFormats from 'ajv-formats'
+import { read_yaml } from "../helpers";
+
+export default class StoryValidator {
+  private static readonly SCHEMA_FILE = path.resolve("./json_schemas/test_story.schema.yaml")
+  private readonly ajv: Ajv2019
+  private readonly validate_schema: ValidateFunction
+
+  constructor() {
+    this.ajv = new Ajv2019({ allErrors: true, strict: false })
```

Comment:
We use plain `ajv` elsewhere, why is this using a different one?

Reply:
It's because of the use of [unevaluatedProperties](https://github.com/opensearch-project/opensearch-api-specification/pull/362/files#diff-a885c8c21d6578474539d576ffef321a7b78bdff03e878868a1e6d37242eece9R26), which is a new feature in JSON Schema draft 2019-09. We use this in `test_story.schema` to enable `additionalProperties: false` for `allOf` objects. This was how I caught [this infraction](https://github.com/opensearch-project/opensearch-api-specification/pull/362/files#diff-016beb89472fc47fd47a2e9c8a9077f02954b509dae821c4c7fb821e26cbc9c6L15-L16).

Reply:
Should we upgrade all schema checks?